kb/development/deprecation.md

```diff
@@ -1,22 +1,25 @@
 # Deprecations
 
-To ensure backwards-compatibility between versions of Mbed TLS, functions in the public interface may not be removed except during the release of a new major version. Instead, functions may be 'deprecated', indicating to users that they will be removed in the future.
+To ensure backwards-compatibility between versions of Mbed TLS, functions or constants in the public interface may not be removed except during the release of a new major version. Instead, they may be 'deprecated', indicating to users that they will be removed in the future.
+
+When a build is made with `MBEDTLS_DEPRECATED_WARNING` defined, a compiler warning will be generated to warn a user that the function or constant will be removed at some point in the future, notifying users that they should change to a replacement function or constant at their own convenience. To find examples of past deprecations, look at the previous major version of Mbed TLS.
+
+## Deprecating functions
 
 To deprecate a function, the following actions must be taken:
 * In the header file (e.g. `include/mbedtls/xyz.h`):
-    * Make sure the header file includes `mbedtls/platform_util.h` (this contains the definition of `MBEDTLS_DEPRECATED`).
     * Surround the function declaration with `#if !defined(MBEDTLS_DEPRECATED_REMOVED)`.
     * Add a line to the docstring consisting of the keyword `\deprecated` followed by the reason for deprecation and (where applicable) the function it is superseded by.
     * Add the `MBEDTLS_DEPRECATED` annotation to the function prototype (so `int foo();` becomes `int MBEDTLS_DEPRECATED foo();`).
 * In the source file (e.g. `library/xyz.c`):
+    * Make sure the header file includes `mbedtls/platform_util.h` (this contains the definition of `MBEDTLS_DEPRECATED`).
```

Comment:
This should stay under the “header file” first-level bullet point.

You don't need to include `mbedtls/platform_util.h` in the C file (but it'll be included indirectly anyway since it's needed in the H file and the H file needs to be included in the C file).

Reply:
Whoops, I slipped up there - thanks!